src/main/java/com/google/jenkins/plugins/computeengine/InstanceConfiguration.java

```diff
@@ -339,12 +339,11 @@ public Instance instance() throws IOException {
         if (StringUtils.isNotEmpty(template)) {
             // TODO: JENKINS-55285
             InstanceTemplate instanceTemplate = cloud.client.getTemplate(nameFromSelfLink(cloud.projectId), nameFromSelfLink(template));
-            Map<String, String> templateLabels = instanceTemplate.getProperties().getLabels();
-            Map<String, String> mergedLabels = new HashMap<>(templateLabels);
-            mergedLabels.putAll(googleLabels);
-            instance.setLabels(mergedLabels);
+            if (instanceTemplate.getProperties().getLabels() != null) {
+                Map<String, String> templateLabels = instanceTemplate.getProperties().getLabels();
+                appendLabels(templateLabels);
```

Comment:
You cannot do that like that cause your changing Instance configuration for good..
And you will add labels with every run..

Reply:
Great catch, I've made a change. Feel free to comment.